docs/plugins/websocket.md

```diff
@@ -29,67 +29,122 @@ export class AppModule {}
-  constructor(public animals: any[]) {}
+export class AddMessage {
+  static type = '[Chat] Add message';
+  constructor(public from: string, public message: string) {}
 }
 ```
 
-Now in our state, we just bind to the `AddAnimals` action like any other 
-action:
+Assume we've got some `messages` state where we store our chat messages:
 
 ```TS
-@State<ZooStateModel>({
-  defaults: {
-    animals: []
-  }
+export interface Message {
+  from: string;
+  message: string;
+}
+
+@State<Message[]>({
+  name: 'messages',
+  defaults: []
 })
-export class ZooState {
-  @Action(AddAnimals)
-  addAnimals(ctx: StateContext<ZooStateModel>, action: AddAnimals) {
-    ctx.setState({ animals: [...action.animals] });
+export class MessagesState {
+  @Action(AddMessage)
+  addMessage(ctx: StateContext<Message[]>, { from, message }: AddMessage) {
+    const state = ctx.getState();
+    // omit `type` property that server socket sends
+    ctx.setState([...state, { from, message }]);
   }
 }
 ```
 
-To send messages to the server, we can dispatch the `SendWebSocketMessage` with
-the payload being what you want to send.
+We are able to send messages to the server by dispatching the `SendWebSocketMessage` with
+the payload that you want to send to the server. Let's try it out:
 
 ```TS
 @Component({ ... })
 export class AppComponent {
 
   constructor(private store: Store) {}
 
-  onClick() {
-    this.store.dispatch(new SendWebSocketMessage({ foo: true }));
+  sendMessage(from: string, message: string) {
+    const event = new SendWebSocketMessage({
+      type: 'message',
+      from,
+      message
+    });
+
+    this.store.dispatch(event);
   }
 
 }
 ```
 
-When sending the message, remember the send is accepting a JSON-able object.
+When sending the message, remember the send is accepting a JSON-able object. The socket on the server side is able to listen to the `message` event. To clarify things out imagine such code:
+
+```TS
+const { Server } = require('ws');
+const { createServer } = require('http');
+
+const app = require('express')();
+
+const server = createServer(app);
+const ws = new Server({ server });
+
+server.listen(4200);
+
+ws.on('connection', (socket) => {
+  socket.on('message', (data) => {
+    // That's the object that we passed into `SendWebSocketMessage` constructor
+    const { type, from, message } = JSON.parse(data);
+
+    if (type === 'message') {
+      const event = JSON.stringify({
+        type: '[Chat] Add message',
+        from,
+        message
+      });
+
+      // That's the same as `broadcast`
+      // we want to send message to all connected
+      // to the chat clients
+      ws.clients.forEach((client) => {
+        client.send(event);
+      });
+    }
+  });
+});
+```
+
+Notice that you have to specify `type` property on server side, otherwise you will get an error - `Type ... not found on message`. If you don't wanna use the `type` key - you can specify your own when calling `forRoot`:
```

Comment:
```suggestion
Notice that you have to specify `type` property on server side, otherwise you will get an error - `Type ... not found on message`. If you don't want to use a property called `type` as the key then you can specify your own property name when calling `forRoot`:
```